### PR TITLE
Update static-channel-backup.md

### PR DESCRIPTION
**Pull request by the-jade-rabbitt**
#### What

The recommended scb-backup script in v3 RaspiBolt guide fails with "No such file or directory"
I am still at the point in the guide of running this script with both local and remote backup methods disabled.

### Why

tilde character in the path does not work inside quotes.

#### How

I'm suggesting replacing ~ with $HOME which is more universal and avoids the error in this case.
You could just remove the quotes but that wouldn't work where there are spaces in the string assignments.

#### Scope

- [ ] significant change to core configuration
- [ ] independent bonus guide
- [x] simple bug fix

#### Test & maintenance

I no longer get the error when I run the script as user lnd with my proposed changes.
The scb-backup.service status now shows as running (it was getting 'run too frequently' before)

### CODE REVIEW

**Review by VajraOfIndra — COMMENTED** [review]
tACK, I tested it by using `$HOME` and simulated a SCB change, all good.

I'll let someone more expert in shell script review and approve the change, but as far as I can tell this solves a bug in the guide! 

Thanks @the-jade-rabbitt !

**Comment by the-jade-rabbitt** [comment]
Thanks @VajraOfIndra 
I came across another bug in this script when I enabled remote and local backups.
The file names were not being updated with current timestamps because they were only being set once at script startup.  This caused the script to bomb out during the remote backup and resulted in the system service restarting.  The effect of this was only half the backups were making it to the remote backup! 
I've pushed an amend to the existing commit this time.
This will need further review from somebody since I've changed the PR since you last tested it above.
I tested it by touching the watched file while watching the journal log (as recommended by the guide) and also checked the changes to the remote and local backups and I think I can see it working as intended with my fixes in there.

**Comment by VajraOfIndra** [comment]
> Thanks @VajraOfIndra I came across another bug in this script when I enabled remote and local backups. The file names were not being updated with current timestamps because they were only being set once at script startup. This caused the script to bomb out during the remote backup and resulted in the system service restarting. The effect of this was only half the backups were making it to the remote backup! I've pushed an amend to the existing commit this time. This will need further review from somebody since I've changed the PR since you last tested it above. I tested it by touching the watched file while watching the journal log (as recommended by the guide) and also checked the changes to the remote and local backups and I think I can see it working as intended with my fixes in there.

@Stadicus @RooSoft: I don't feel I know enough about bash scripting, could you review these two changes? The SCB guide does not work as it is right now so it's be great to merge a fix asap.

**Comment by RooSoft** [comment]
`$HOME` and `~` are pretty much synonyms...

IMO `~` is way easier to read. I personally never use `$HOME`.

**Comment by RooSoft** [comment]
One thing I have to mention, `~` is a relative path...

In my experience, `systemd` and relative paths don't bode well together... maybe that's why @the-jade-rabbitt proposed $HOME, as that variable can be set in environment variables of the systemd config file, while I doubt it's possible with `~`.

**Comment by the-jade-rabbitt** [comment]
Hi @RooSoft, the script produces an error if you use a ~ inside quotes because it is then treated as a literal.
Here is a stackoverflow that mensions this issue.
https://unix.stackexchange.com/questions/151850/why-doesnt-the-tilde-expand-inside-double-quotes
It does work if you remove the quotes but I don't think it can be avoided where there are spaces within the string assignments.

Using an absolute path would work - but I think it causes a problem with permissions unless you go through the lnd user path with the simlinks.

**Review by RooSoft — CHANGES_REQUESTED** [review]
I like the way that variables are now defined closer to where they're useful.

To further improve to this regard, functions such as `run_local_backup_on_change` and `run_remote_backup_on_change` should accept variables... as code is laid out actually, these functions are defined before the variables they use. It doesn't matter much as they'll probably be called once the variables exist, but function arguments would simplify things up and avoid errors on future changes.

Unsure about this, but I think being more specific by setting up the $HOME environment variable in the systemd config file might help on execution.

**Comment by RooSoft** [comment]
> Using an absolute path would work - but I think it causes a problem with permissions unless you go through the lnd user path with the simlinks.

Regarding permission rights, absolute and relative paths are the same... the difference is that in certain circumstances, relatives paths don't translate to absolute. As far as I know, systemd requires absolute paths, even when the user is specified...

**Comment by Stadicus** [comment]
Updating the timestamps shortly before these are needed is certainly a good thing. :) Initiating them at the beginning can't work of course. Thanks for bringing this up.

But I also think all "configuration" variables should be defined at the top of a script. For example, I'd avoid hardcoding the `/mnt/static-channel-backup-external...` path in the middle of the execution code.

That brings me to another question: why would we want to go through the user's home directory in the first place? We can keep it simple and simply specify three absolute directories in the beginning, easy to change to any system configuration:

* `SCB_SOURCE_FILE="/data/lnd/data/chain/bitcoin/mainnet/channel.backup"`
* `LOCAL_BACKUP_DIR="/mnt/static-channel-backup-external/"`
* `REMOTE_BACKUP_DIR="/data/lnd/remote-lnd-backup/"`

Then, during runtime, the file variables can be created from these, without hardcoding any specific directories. I feel explicit is usually better and easier to understand.

**Comment by the-jade-rabbitt** [comment]
Yes - much better. PR updated.  Thanks!

I've really enjoyed going through the guide.  Got everything up and running on my pi now.
I've also added a service to run testnet too.
Next steps for me will be Learning Bitcoin from the Command Line with Christopher Allen :)

**Comment by the-jade-rabbitt** [comment]
oops, removing the $HOME now ...

**Comment by RooSoft** [comment]
@Stadicus I agree about not relying on users homes for that kind of job...

File: static-channel-backup.md [hunk 1]
```diff
@@ -129,16 +129,16 @@ We create a shell script to monitor `channel.backup` and make a copy to our back
...
   # Locations of source SCB file, formatted backup files and Git repo
-  SOURCEFILE="~/.lnd/data/chain/bitcoin/mainnet/channel.backup"
-  LOCAL_BACKUP_FILE="/mnt/static-channel-backup-external/channel-$(date +"%Y%m%d-%H%M%S").backup"
-  GITREPO="~/.lnd/remote-lnd-backup"
-  REMOTE_BACKUP_FILE="~/.lnd/remote-lnd-backup/channel-$(date +"%Y%m%d-%H%M%S").backup"
-
+  SCB_SOURCE_FILE="/data/lnd/data/chain/bitcoin/mainnet/channel.backup"
+  LOCAL_BACKUP_DIR="/mnt/static-channel-backup-external"
+  REMOTE_BACKUP_DIR="/data/lnd/remote-lnd-backup"
+  GITREPO="/data/lnd/remote-lnd-backup"
```
Comment (Stadicus) [inline static-channel-backup.md:135]: I think we could get rid of `GITREPO`, as it's redundant:

* Delete this line here
* Replace it with `REMOTE_BACKUP_DIR` for the `cd` command in line 148

File: static-channel-backup.md [hunk 2]
```diff
@@ -161,17 +162,20 @@ We create a shell script to monitor `channel.backup` and make a copy to our back
...
+        inotifywait $SCB_SOURCE_FILE
         echo "channel.backup has been changed!"
 
+        LOCAL_BACKUP_FILE="$LOCAL_BACKUP_DIR/channel-$(date +"%Y%m%d-%H%M%S").backup"
+        REMOTE_BACKUP_FILE="$REMOTE_BACKUP_DIR/channel-$(date +"%Y%m%d-%H%M%S").backup"
+
         if [ "$LOCAL_BACKUP_ENABLED" == true ]; then
           echo "Local backup is enabled"
-          run_local_backup_on_change
+          run_local_backup_on_change $LOCAL_BACKUP_FILE
```
Comment (Stadicus) [inline static-channel-backup.md:173]: Being pedantic, the variable should be in double quotes:
```
run_local_backup_on_change "$LOCAL_BACKUP_FILE"
```
https://github.com/koalaman/shellcheck/wiki/SC2086

File: static-channel-backup.md [hunk 3]
```diff
@@ -129,16 +129,16 @@ We create a shell script to monitor `channel.backup` and make a copy to our back
   REMOTE_BACKUP_ENABLED=false
 
   # Locations of source SCB file, formatted backup files and Git repo
```
Comment (Stadicus) [inline static-channel-backup.md:131]: Could be slightly rephrased to reflect the new variables:

```
# Locations of source SCB file and the backup target directories (local and remote)
```

**Comment by Stadicus** [comment]
> Next steps for me will be Learning Bitcoin from the Command Line with Christopher Allen :)

That's an awesome resource! Never quite made it through it, though... :smile: 

> Would it have been better if I pushed a sequence of commits to this PR as opposed to ammending each time?

Doesn't really matter, no worries. I configured the repository so that it automatically squashes all commits on merge anyway. Thanks for contributing in the first place, much appreciated! :pray:

**Comment by the-jade-rabbitt** [comment]
> I configured the repository so that it automatically squashes all commits on merge anyway 

👍

If the automatic squash loses the 1st commit message I can update it if you like.
Or does it keep them all?

File: static-channel-backup.md (hunk 2 above)
Comment (the-jade-rabbitt) [reply static-channel-backup.md]: Today I leaned something :)

**Review by Stadicus — APPROVED** [review]
tACK

**Comment by Stadicus** [comment]
> If the automatic squash loses the 1st commit message I can update it if you like.
> Or does it keep them all?

It just keeps all the commit titles as a list, without the individual commit messages. If you want, you can phrase a commit message here so that I can copy/paste it when I'll merge after @RooSoft's review

**Comment by the-jade-rabbitt** [comment]
The titles will be fine :-)

**Comment by RooSoft** [comment]
Seems to me there's a conflict... maybe I could review the PR once it's resolved...

**Comment by the-jade-rabbitt** [comment]
> Seems to me there's a conflict... maybe I could review the PR once it's resolved...

Thanks @RooSoft, I've rebased it now.

**Review by RooSoft — APPROVED** [review]
That's ok to me, but I'd still make arguments instead of using environment variables in functions.

Might propose a PR later to change this if this idea proves to be popular. Would make things more obvious in my opinion.

**Comment by Stadicus** [comment]
@the-jade-rabbitt, your change is now live. Thanks for contributing! :pray:

**Pull request merged**